src/firebase.js

```diff
@@ -281,7 +287,13 @@
 
           function init() {
             ref.on('value', applyUpdate, error);
-            ref.once('value', function() { resolve(null); }, resolve);
+            ref.once('value', function(snap) {
+              if (angular.isArray(snap.val())) {
+                throw new Error('Storing data using array indices in Firebase can result in unexpected behavior. See https://www.firebase.com/docs/rest/guide/understanding-data.html#section-arrays-in-firebase for more information.');
```

Comment:
Do I read this code correctly? This will allow the bindings to work, but prevent the $loaded() promise from resolving or being rejected? I think we have access to $log here. If we want to allow the bindings to continue functioning, I'd prefer a $log.warn() or $log.error() to actually throwing from here, so $loaded still works as well.